### PR TITLE
[#174034404] Refactory CosmosDB SDK v3 (keeps cosmos bindings)

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

File: DeleteEligibilityCheckActivity/handler.ts [hunk 1]
```diff
@@ -49,27 +49,25 @@ export const getDeleteEligibilityCheckActivityHandler = (
...
-              );
-              throw queryError;
-            },
-            id => fromEither(right(id))
-          )
+        eligibilityCheckModel.deleteOneById(fiscalCode).foldTaskEither(
+          err => {
+            if (
+              err.kind === "COSMOS_ERROR_RESPONSE" &&
+              err.error.code === 404
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply DeleteEligibilityCheckActivity/handler.ts:56]: I'm verified locally and when a document is not found `deleteOneById` returns a `CosmosErrors` with `error.code=404`

File: models/user_bonus.ts [hunk 2]
```diff
@@ -20,69 +26,49 @@ const UserBonus = t.interface({
...
+    const iterator = this.getQueryIterator({
+      parameters: [
+        {
+          name: "@fiscalCode",
+          value: fiscalCode
+        }
+      ],
+      query: `SELECT * FROM m WHERE m.${USER_BONUS_MODEL_PK_FIELD} = @fiscalCode`
+    })[Symbol.asyncIterator]();
+    return fromEither(tryCatch2v(() => iterator, toCosmosErrorResponse));
```
Comment (gunzip) [inline models/user_bonus.ts:72]: this looks odd

File: package.json [hunk 3]
```diff
@@ -58,14 +57,14 @@
...
+    "@azure/cosmos": "^3.7.4",
     "applicationinsights": "^1.7.5",
     "azure-storage": "^2.10.3",
     "date-fns": "^2.14.0",
-    "documentdb": "^1.15.3",
     "durable-functions": "^1.4.1",
     "express": "^4.15.3",
     "fp-ts": "1.17.4",
-    "io-functions-commons": "^10.5.0",
+    "io-functions-commons": "^12.10.0",
```
Comment (gunzip) [inline package.json:67]: 12.11.0

File: models/user_bonus.ts (hunk 2 above)
Comment (BurnedMarshal) [reply models/user_bonus.ts]: Fixed in 0c17de0

File: package.json (hunk 3 above)
Comment (BurnedMarshal) [reply package.json]: fixed in 970a349

**Comment by gunzip** [comment]
closing in favor of #159

**Pull request closed without merging**